src/main/java/org/seqdoop/hadoop_bam/AnySAMInputFormat.java

```diff
@@ -229,18 +230,21 @@ else if (split instanceof FileVirtualSplit)
 		final List<InputSplit> origSplits =
 				BAMInputFormat.removeIndexFiles(super.getSplits(job));
 
+		final List<InputSplit> sortedSplits = new ArrayList<>(origSplits);
+		sortedSplits.sort(Comparator.comparing(split -> ((FileSplit) split).getPath()));
```

Comment:
Is this cast problematic?  I ran into the problem that some splits are `FileVirtualSplits` which are not a subtype of `FileSplit`, there isn't any common super type that offers `getPath`.  Maybe `FileVirtualSplit` should extend `FileSplit`?

Reply:
`AnySAMInputFormat` extends Hadoop's `FileInputFormat`, which returns `FileSplit` objects and never `FileVirtualSplit`. So the cast is safe at this point in the code.